AnalogSensor.cpp

```diff
@@ -1,164 +1,38 @@
- * @param data (int) The data to be rescaled.
- * @return (int) The rescaled data.
- */
-int AnalogSensor::rescale(int data) {
-    //Transform data
-    return data;
-};
-
-
-
-/**
- * @brief Check if the sensor is ready for the next update.
- *
- * Determines if the elapsed time since the last update is greater than or equal to 
- * the specified wait time.
- *
- * @return (bool) True if the sensor is ready for the next update, false otherwise.
- */
-bool AnalogSensor::readyToCheck() {
-    //millis gets arduino time
-    return (waitTime <= int(millis() - previousUpdateTime));
-};
-
-/**
- * @brief Get the unique identifier for the sensor.
- * @return (int) The sensor identifier.
- */
-int AnalogSensor::getId() {
-    return sensorID;
-}
-
-/**
- * @brief Get the input pin of the analog sensor.
- * @return (int) The input pin number.
- */
-int AnalogSensor::getPins() {
-    return inputPins[0];
+AnalogSensor::AnalogSensor(const uint32_t id, const bool criticality, const uint8_t pin, const uint32_t readInterval)
+{
+    this->id = id;
+    this->criticality = criticality;
+    this->readInterval = readInterval;
+    this->pin = pin;
 }
 
-/**
- * @brief Get the update frequency wait time.
- * @return (int) The update frequency wait time in milliseconds.
- */
-int AnalogSensor::getWaitTime() {
-    return waitTime;
-}
-
-/**
- * @brief Get the length of the data array.
- * @return (int) The length of the data array.
- */
-int AnalogSensor::getDataLength() {
-    return dataLength;
-}
+SensorData AnalogSensor::read()
+{
+    lastRead = millis();
+    SensorData sensorData = SensorData(id, 1);
 
-/**
- * @brief Set the input pin for the analog sensor.
- * @param inPin (int) The new input pin number.
- */
-void AnalogSensor::setPin(int inPin, int index=0) {
-    inputPins[index] = inPin;
-}
+    uint8_t buf[sizeof(int)];
+    ByteConverter::intToBytes(analogRead(pin), buf);
+    sensorData.setMsg(buf, 4);
 
-/**
- * @brief Set the update frequency wait time.
- * @param inWait (int) The new update frequency wait time in milliseconds.
- */
-void AnalogSensor::setWaitTime(int inWait) {
-    waitTime = inWait;
+    return sensorData;
 }
 
-/**
- * @brief Set the unique identifier for the sensor.
- * @param inId (int) The new sensor identifier.
- */
-void AnalogSensor::setId(int inId) {
-    sensorID = inId;
-}
+Pot::Pot(
+    const uint32_t id, const bool criticality, const uint8_t pin, const uint32_t readInterval, const char* name
+) : AnalogSensor(id, criticality, pin, readInterval), name(name) {  }
 
-bool AnalogSensor::getCritical() {
-    return isCritical;
+void Pot::debugPrint(const CAN_message_t& canMsg) const
+{
+    Serial.print(name);
+    Serial.println(" Potentiometer CAN Message:");
+    Serial.print("Timestamp: ");
+    Serial.println(canMsg.timestamp);
+    const int value = ByteConverter::intFromBytes(canMsg.buf);
+    Serial.print("Value: ");
+    Serial.println(value);
+    Serial.println();
 }
```

Comment:
A whole class for a pretty print seems kinda overkill to me. Would it not be easier to put it as part of the parent and just print the ID? I think that would be more clear cause we would know the exact sensor it was instead of "potentiometer"

Reply:
Agreed - I am going to remove the Pot and Switch classes entirely.